src/codeswitch/data_generation/synthatic_data_generation.py

```diff
@@ -0,0 +1,160 @@
+import os
+import random
+import re
+import time
+from collections import Counter
+
+import pandas as pd
+from openai import OpenAI
+from tqdm import tqdm
+
+# pylint: disable=E1101
+
+# Initialize the OpenAI client
+api_key = os.getenv("open_ai_key")
+if not api_key:
+    raise ValueError("The open_ai_key environment variable is not set.")
+
+client = OpenAI(api_key=api_key)
+
+
+def classify_topic(utterance, retries=3, delay=2):
+    for _ in range(retries):
+        try:
+            stream = client.chat.completions.create(
```

Comment:
You should use json mode so youre sure the results can be parsed and without comments. Refer to this file on how to do it: https://github.com/cheyannelam/code-switch/blob/0bf00dfe8c0801765534b4cb456b0ddc813738b5/src/codeswitch/data_generation/utterance_generation.py